src/grpc/grpc_server.cc

```diff
@@ -2435,6 +2435,139 @@ Server::Create(
   return nullptr;  // success
 }
 
+TRITONSERVER_Error*
+Server::Create(
+    std::shared_ptr<TRITONSERVER_Server>& server, UnorderedMapType& data,
+    const RestrictedFeatures& restricted_features,
+    std::unique_ptr<Server>* service)
+{
+  Options grpc_options;
+
+  RETURN_IF_ERR(GetOptions(grpc_options, data));
+
+  TRITONSERVER_Error* err =
+      Create(server, nullptr, nullptr, grpc_options, service);
+  return err;
+}
+
+TRITONSERVER_Error*
+Server::GetOptions(Options& options, UnorderedMapType& data)
+{
+  SocketOptions socket_selection;
+  SslOptions ssl_selection;
+  KeepAliveOptions keep_alive_selection;
+
+  RETURN_IF_ERR(GetSocketOptions(socket_selection, data));
+  RETURN_IF_ERR(GetSslOptions(ssl_selection, data));
+  RETURN_IF_ERR(GetKeepAliveOptions(keep_alive_selection, data));
+
+  int infer_compression_level_key;
+  int infer_allocation_pool_size;
+  std::string forward_header_pattern;
+
+  RETURN_IF_ERR(
+      get_value(data, "infer_compression_level", infer_compression_level_key));
+
+  RETURN_IF_ERR(get_value(
+      data, "infer_allocation_pool_size", infer_allocation_pool_size));
+  RETURN_IF_ERR(
+      get_value(data, "forward_header_pattern", forward_header_pattern));
+
+  options.socket_ = socket_selection;
+  options.ssl_ = ssl_selection;
+  options.keep_alive_ = keep_alive_selection;
+  options.infer_compression_level_ =
+      static_cast<grpc_compression_level>(infer_compression_level_key);
+  options.infer_allocation_pool_size_ = infer_allocation_pool_size;
+  options.forward_header_pattern_ = forward_header_pattern;
+
+  return nullptr;
+}
+
+TRITONSERVER_Error*
+Server::GetSocketOptions(SocketOptions& options, UnorderedMapType& data)
+{
+  std::string address;
+  int port;
+  int reuse_port;
+  RETURN_IF_ERR(get_value(data, "address", address));
+  RETURN_IF_ERR(get_value(data, "port", port));
+  RETURN_IF_ERR(get_value(data, "reuse_port", reuse_port));
+
+  options.address_ = address;
+  options.port_ = port;
+  options.reuse_port_ = reuse_port;
+
+  return nullptr;
+}
+
+TRITONSERVER_Error*
+Server::GetSslOptions(SslOptions& options, UnorderedMapType& data)
+{
+  int use_ssl;
+  std::string server_cert;
+  std::string server_key;
+  std::string root_cert;
+  int use_mutual_auth;
+
+  RETURN_IF_ERR(get_value(data, "use_ssl", use_ssl));
+  RETURN_IF_ERR(get_value(data, "server_cert", server_cert));
+  RETURN_IF_ERR(get_value(data, "server_key", server_key));
+  RETURN_IF_ERR(get_value(data, "root_cert", root_cert));
+  RETURN_IF_ERR(get_value(data, "use_mutual_auth", use_mutual_auth));
+
+  options.use_ssl_ = use_ssl;
+  options.server_cert_ = server_cert;
+  options.server_key_ = server_key;
+  options.root_cert_ = root_cert;
+  options.use_mutual_auth_ = use_mutual_auth;
```

Comment:
nit: You can set options args directly
```cpp
  RETURN_IF_ERR(get_value(data, "use_ssl", options.use_ssl_));
```

Reply:
Same for all the get helper functions that follow this pattern

Reply:
This could definitely reduce the size of the helper functions, but might cause errors for the boolean options like `use_ssl`. Additionally, `compression_level` is represented as an integer in the dictionary passed from the python dataclass, so it needs to be casted back to type `grpc_compression_level`.

Reply:
Changes are implemented.